packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/utils/generate-auth-trigger-template.test.ts

```diff
@@ -1,34 +1,76 @@
-import { Construct } from '@aws-cdk/core';
-import { generateNestedAuthTriggerTemplate, createCustomResourceForAuthTrigger, CustomResourceAuthStack } from '../../../../provider-utils/awscloudformation/utils/generate-auth-trigger-template';
+import {
+  AuthTriggerConnection,
+  AuthTriggerPermissions,
+  TriggerType,
+} from '../../../../provider-utils/awscloudformation/service-walkthrough-types/cognito-user-input-types';
+// eslint-disable-next-line spellcheck/spell-checker
+import { createCustomResourceforAuthTrigger } from '../../../../provider-utils/awscloudformation/utils/generate-auth-trigger-template';
 
-jest.mock('uuid', () => {
-  return {
-    v4: () => {
-      return '4fd798a7-86d3-46c2-81e1-40eed9df38ec';
-    }
-  };
+jest.mock('uuid');
+describe('generate Auth Trigger Template', () => {
+  it('successfully generate auth Trigger Template', async () => {
+    const mockAuthTriggerConnections: AuthTriggerConnection[] = [
+      {
+        lambdaFunctionName: 'randomFn',
+        triggerType: TriggerType.PostConfirmation,
+        lambdaFunctionArn: 'randomArn',
+      },
+    ];
+
+    const mockAuthTriggerPermissions: AuthTriggerPermissions[] = [
+      {
+        policyName: 'AddToGroupCognito',
+        trigger: 'PostConfirmation',
+        effect: 'Allow',
+        actions: ['cognito-idp:AdminAddUserToGroup', 'cognito-idp:GetGroup', 'cognito-idp:CreateGroup'],
+        resource: {
+          paramType: '!GetAtt',
+          keys: ['UserPool', 'Arn'],
+        },
+      },
+    ];
+    // eslint-disable-next-line spellcheck/spell-checker
+    const cfn = await createCustomResourceforAuthTrigger(mockAuthTriggerConnections, false, mockAuthTriggerPermissions);
+    expect(cfn).toMatchSnapshot();
```

Comment:
If the intent here is to check that the auth trigger is working as expected, a better test would be unit tests around the actual functionality of the lambda code, not just checking that it matches a snapshot. Could these tests be removed in favor of test(s) like that?

Reply:
I have changed the logic to check for IAM permissions added to template or not